### PR TITLE
Cover/cmds/core/sync

### PR DESCRIPTION
**Pull request by llogen**
Refactored cmds/core/sync and added tests.

### CODE REVIEW

File: cmds/core/sync/sync.go [hunk 1]
```diff
@@ -28,39 +30,42 @@ var (
...
+	for _, fileName := range args {
 		f, err := os.OpenFile(fileName, syscall.O_RDONLY|syscall.O_NOCTTY|syscall.O_CLOEXEC, 0o644)
 		if err != nil {
-			log.Fatal(err)
+			return err
 		}
 		_, _, err = syscall.Syscall(syscallNum, uintptr(f.Fd()), 0, 0)
-		if err != nil {
-			log.Fatal(err)
+		if errors.Is(err, fs.ErrNotExist) {
```
Comment (hugelgupf) [inline cmds/core/sync/sync.go:46]: why change from `err != nil` to `errors.Is(err, fs.ErrNotExist)`? please change back?

File: cmds/core/sync/sync_test.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	"testing"
+
+	"github.com/u-root/u-root/pkg/testutil"
+)
+
+func TestSync(t *testing.T) {
+	testutil.SkipIfNotRoot(t)
+	d, err := os.MkdirTemp(os.TempDir(), "sync")
+	if err != nil {
+		t.Errorf("failed to create tmp folder")
```
Comment (hugelgupf) [inline cmds/core/sync/sync_test.go:19]: report error

t.Errorf("Failed to create tmp folder: %v", err)

File: cmds/core/sync/sync_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	file1, err := os.CreateTemp(d, "file1")
+	if err != nil {
+		t.Errorf("failed to create tmp file1")
```
Comment (hugelgupf) [inline cmds/core/sync/sync_test.go:23]: ditto

File: cmds/core/sync/sync_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	file2, err := os.CreateTemp(d, "file2")
+	if err != nil {
+		t.Errorf("failed to create tmp file2")
```
Comment (hugelgupf) [inline cmds/core/sync/sync_test.go:27]: ditto

File: cmds/core/sync/sync.go (hunk 1 above)
Comment (llogen) [reply cmds/core/sync/sync.go:46]: syscall.Syscall gives back an errno err which can't be nil. So i wanted to check if there is no error.
But it has to be: if !errors.Is(err, fs.ErrNotExist)
Please correct me if i am wrong.

**Review by walterchris — APPROVED** [review]

**Pull request merged**